arrows/ocv/register_algorithms.cxx

```diff
@@ -71,84 +70,288 @@ namespace kwiver {
+    ;
 
-  REGISTER_TYPE( ocv::extract_descriptors_BRISK );
-  REGISTER_TYPE( ocv::extract_descriptors_ORB );
 
-  REGISTER_TYPE( ocv::match_features_bruteforce );
-  REGISTER_TYPE( ocv::match_features_flannbased );
+  fact = vpm.ADD_ALGORITHM( "ocv_ORB", kwiver::arrows::ocv::detect_features_ORB );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                       "OpenCV feature detection via the ORB algorithm" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "ocv_simple_blob", kwiver::arrows::ocv::detect_features_simple_blob );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                       "OpenCV feature detection via the simple_blob algorithm." )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "ocv_BRISK", kwiver::arrows::ocv::extract_descriptors_BRISK );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                       "OpenCV feature-point descriptor extraction via the BRISK algorithm" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "ocv_ORB", kwiver::arrows::ocv::extract_descriptors_ORB );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                       "OpenCV feature-point descriptor extraction via the ORB algorithm" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "ocv_brute_force", kwiver::arrows::ocv::match_features_bruteforce );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "ocv_flann_based", kwiver::arrows::ocv::match_features_flannbased );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
```

Comment:
Use this

    OpenCV feature matcher using FLANN (Approximate Nearest Neighbors)